sdks/js/ditto-chat-core/README.md

```diff
@@ -233,6 +306,65 @@ The chat system triggers notifications for various events:
 - New messages in subscribed rooms
 - User mentions
 
+## Singleton Store Pattern
+
+**Important:** DittoChatCore uses a **global singleton pattern** to ensure that all npm packages share the same Zustand store instance.
+
+### Why This Matters
+
+When `@dittolive/ditto-chat-core` is installed in multiple packages (e.g., in your main app and in `@dittolive/ditto-chat-ui`), each package could potentially get its own copy of the module. To prevent multiple independent store instances, we use `globalThis` to maintain a single shared store.
+
+### What This Means for You
+
+1.  **Initialize once** - Call `useDittoChat()` in your root component
+2.  **Use anywhere** - Call `useDittoChatStore()` in any component from any package
+3.  **Shared state** - All packages automatically share the same state
+
+### Example
+
+```typescript
+// In your main app (initialize once)
+function App() {
+  useDittoChat({
+    ditto: dittoInstance,
+    userId: currentUserId,
+    userCollectionKey: userCollectionKey,
+  })
+
+  return <YourApp />
+}
+
+// In any component, from any package (use the shared store)
+function ChatComponent() {
+  const messages = useDittoChatStore((state) => state.messagesByRoom)
+  // All components see the same state
+}
+```
+
+### Ensuring Single Zustand Installation
+
+To prevent multiple copies of Zustand, add this to your **main application's** `package.json`:
+
+```json
+{
+  "overrides": {
+    "zustand": "^5.0.8"
+  }
+}
+```
+
+Then verify with:
+
+```bash
+npm ls zustand
+```
```

Comment:
I think this can be removed, too. The peer dependencies should be enough to inform a developer.